source/MaterialXRuntime/RtConnectableApi.h

```diff
@@ -0,0 +1,66 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_RTCONNECTABLEAPI_H
+#define MATERIALX_RTCONNECTABLEAPI_H
+
+#include <MaterialXRuntime/Library.h>
+#include <MaterialXRuntime/RtAttribute.h>
+#include <MaterialXRuntime/RtRelationship.h>
+
+/// @file
+/// TODO: Docs
+
+namespace MaterialX
+{
+
+class RtConnectableApi;
+using RtConnectableApiPtr = shared_ptr<RtConnectableApi>;
+
+/// @class RtConnectableApi
+/// API for validating attribute connections and relationships for a prim type.
+class RtConnectableApi
```

Comment:
This is the new API for connectability. With this API we can override per prim schema which connections and relationships are valid for the prim type.